rclpy/src/rclpy_common/src/common.c

```diff
@@ -338,3 +338,100 @@ rclpy_convert_to_py(void * message, PyObject * pyclass)
   }
   return convert(message);
 }
+
+PyObject *
+_rclpy_convert_to_py_topic_info(const rmw_topic_info_t * topic_info)
+{
+  PyObject * py_qos_profile = NULL;
+  PyObject * py_node_name = NULL;
+  PyObject * py_node_namespace = NULL;
+  PyObject * py_topic_type = NULL;
+  PyObject * py_gid = NULL;
+  PyObject * py_topic_info_dict = NULL;
+
+  const rmw_qos_profile_t qos_profile = topic_info->qos_profile;
+  py_qos_profile = rclpy_common_convert_to_qos_dict(&qos_profile);
+  if (!py_qos_profile) {
+    goto fail;
+  }
+  const char * node_name = topic_info->node_name;
```

Comment:
nit: avoid unnecessary variable definitions (doesn't `goto` complain about crossing a variable definition?).

Reply:
Is there a reason for this preference? I am just curious to know. 
I've removed using them anyway.

Reply:
It's not more readable, it adds more code, and it does the same.
Also, some compilers complain about `goto` statements crossing variable definitions (I'm almost sure it complains in some of our supported platforms, but I'm not completely sure).